### ISSUE TITLE
Modify the admin settings for a bridge IRC room

### ISSUE THREAD

**Issue by bluddy** [matrix-org/matrix-appservice-irc#91]
We have an IRC room #freenode_#ocaml:matrix.org that we'd like to direct people to for ease-of-use. However, the room (accessed via vector.im) doesn't allow for guests currently, and the admin is stated to be @appservice-irc:matrix.org.

Is there any way we can gain admin rights to the room and modify its settings?

**Comment by Kegsay** [comment]
#### The quick answer
There is currently no way for Matrix users to gain ops in that room, because *we don't know* who is allowed to have that power. You can ask me directly (`@kegan:matrix.org`) and I can manually make the bridge allow guest users on your behalf once I've manually joined `#ocaml` on IRC and asked the IRC chan ops for permission.

#### The long answer
This relates heavily to https://github.com/matrix-org/matrix-appservice-irc/issues/46 - and also relates to the fundamental questions about what the purpose of the Matrix room is.

Currently, `#freenode_#channel:matrix.org` rooms are treated as what we call "Portal" rooms - They provide a portal for Matrix users to communicate into the IRC channel as a regular user. They don't own the room. They don't moderate the room. They merely participate like a regular user. The balance of power very much resides on the IRC channel, and the Matrix side purposefully doesn't try to intrude too much into that.

There is another genre of room though which we called "Plumbed" rooms - They are matrix rooms which intend to "plumb" together 1 or more remote networks (e.g. an IRC channel and a Slack channel) in a coherent super-room. In this model, the balance of power very much resides on Matrix (it has to, as different networks have different ideas of "permissions" and only Matrix can serve as the mediator between them). The moderators of the room are Matrix users, and they can control guest access, where to plumb the room into, etc.

So now for the final piece of the puzzle to understand the underlying problem: why aren't all Freenode channels that matrix.org bridges to treated as "Plumbed" rooms? In one word: community. Matrix is new, and will be an intruder onto remote networks like Slack and IRC for quite some time. Typically there are existing communities on these remote networks already and you have 1 or 2 people who try Matrix to bridge everything together. The onus is on those Matrix users to be responsible, but we can't guarantee that - the first user who accesses the remote room via Matrix would have too much power if we just assumed they'd be responsible. That is why I will manually verify with the chan op if "allowing X" is okay. This policy is unlikely to change any time soon.

In the future, we hope to blur those lines somewhat by giving more power to IRC chan ops ( https://github.com/matrix-org/matrix-appservice-irc/issues/46 ), but there are huge impedance mismatches to resolve first.

**Comment by Kegsay** [comment]
> I could see the potential for spam bots to attack IRC channels via matrix without there being any moderation tools available to stop this. 

It wouldn't be worth it. Certainly on the Freenode bridge at least, we use ident so real IRC users can just apply a ban on the user portion of the `user@host`. If it was horrendous, they could easily ban the bridge entirely by banning our `host`. That, plus rate limiting on the Matrix side makes spam from Matrix more hassle than just spinning up new TCP connections to the IRCd itself.

> too much whitespace by default

As in the vector.im interface? Perhaps you should [file a request](https://github.com/vector-im/vector-web/issues) with them :)

> How would we go about creating a 'Plumbed' room to the same IRC channel? Is this possible?

Of course it's possible. Nothing is stopping anyone from running their own Homeserver and IRC bridge and then they can do whatever the hell they want. Just try not to get into a routing loop :)

If you want a plumbed room so you have ops, then it's far easier just to get me to ask the IRC chan ops and we're done. If you want it for more personal reasons, then you can (and some people already have) run your own setup pointed to your own matrix room (which you have ops in). It looks something like this:

```
#freenode_#ocaml:matrix.org
  !roomID:matrix.org             !roomID:your.homeserver
          |                            |
   matrix.org                    your.homeserver
   IRC bridge                      IRC bridge
          |                            |
          +---------#ocaml-------------+
                  (Freenode)
```

Messages sent from `#freenode_#ocaml:matrix.org` would go through to IRC and then be received by *your* IRC bridge and sent through to *your* matrix room, so Matrix users on matrix.org will appear as if they originated from IRC.

**Comment by bluddy** [comment]
Sorry for conflating vector.im and matrix.

My only concern is for setting room options, so if you'd like to go ahead and ask the IRC chan ops, that would be great (if you still need me to message you via IRC about this, let me know). I would also like to have the options set for letting anyone see room history -- no reason not to since this is an open-source project, and to list the room in the matrix directory for easy finding.

**Comment by ara4n** [comment]
i think this is basically solved by #206 ?

**Comment by Kegsay** [comment]
Yes. Closing.
